builtin/providers/softlayer/provider.go

```diff
@@ -0,0 +1,42 @@
+package softlayer
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"username": &schema.Schema{
+				Type: schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.EnvDefaultFunc("SOFTLAYER_USERNAME", nil),
+				Description: "The user name for SoftLayer API operations.",
+			},
+			"api_key": &schema.Schema{
+				Type: schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.EnvDefaultFunc("SOFTLAYER_API_KEY", nil),
+				Description: "The API key for SoftLayer API operations.",
+			},
+		},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"softlayer_virtualserver": resourceSoftLayerVirtualserver(),
```

Comment:
I don't know Softlayer at all so forgive me if this is a silly question, but would you consider calling this resource ``softlayer_virtual_server``, with an underscore separating those words? Is it branded within the Softlayer platform as "Virtualserver" rather than "Virtual Server" or "VirtualServer"?